ratatui-widgets/src/list/rendering.rs

```diff
@@ -12,39 +12,31 @@ use crate::{
 
 impl Widget for List<'_> {
     fn render(self, area: Rect, buf: &mut Buffer) {
-        WidgetRef::render_ref(&self, area, buf);
+        Widget::render(&self, area, buf);
     }
 }
 
-impl WidgetRef for List<'_> {
-    fn render_ref(&self, area: Rect, buf: &mut Buffer) {
+impl Widget for &List<'_> {
+    fn render(self, area: Rect, buf: &mut Buffer) {
         let mut state = ListState::default();
-        StatefulWidgetRef::render_ref(self, area, buf, &mut state);
+        StatefulWidget::render(self, area, buf, &mut state);
     }
 }
 
 impl StatefulWidget for List<'_> {
     type State = ListState;
 
     fn render(self, area: Rect, buf: &mut Buffer, state: &mut Self::State) {
-        StatefulWidgetRef::render_ref(&self, area, buf, state);
+        StatefulWidget::render(&self, area, buf, state);
     }
 }
 
-// Note: remove this when StatefulWidgetRef is stabilized and replace with the blanket impl
 impl StatefulWidget for &List<'_> {
     type State = ListState;
-    fn render(self, area: Rect, buf: &mut Buffer, state: &mut Self::State) {
-        StatefulWidgetRef::render_ref(self, area, buf, state);
-    }
-}
 
-impl StatefulWidgetRef for List<'_> {
-    type State = ListState;
-
-    fn render_ref(&self, area: Rect, buf: &mut Buffer, state: &mut Self::State) {
+    fn render(self, area: Rect, buf: &mut Buffer, state: &mut Self::State) {
```

Comment:
This change broke the doctest case in `render_stateful_widget_ref` in `Frame`:

```
---- ratatui/src/terminal/frame.rs - terminal::frame::Frame<'_>::render_stateful_widget_ref (line 170) stdout ----
error[E0277]: the trait bound `List<'_>: StatefulWidgetRef` is not satisfied
   --> ratatui/src/terminal/frame.rs:185:34
    |
18  | frame.render_stateful_widget_ref(list, area, &mut state);
    |       -------------------------- ^^^^ the trait `StatefulWidgetRef` is not implemented for `List<'_>`
    |       |
    |       required by a bound introduced by this call
    |
note: required by a bound in `Frame::<'_>::render_stateful_widget_ref`
   --> /home/orhun/gh/ratatui/ratatui/src/terminal/frame.rs:191:12
    |
189 |     pub fn render_stateful_widget_ref<W>(&mut self, widget: W, area: Rect, state: &mut W::State)
    |            -------------------------- required by a bound in this associated function
190 |     where
191 |         W: StatefulWidgetRef,
    |            ^^^^^^^^^^^^^^^^^ required by this bound in `Frame::<'_>::render_stateful_widget_ref`

error: aborting due to 1 previous error
```

Btw what's the current status with the blanket implementation of `StatefulWidget for &W where W: StatefulWidget`? I guess we need that here.

Reply:
f8b6cc1be..1e4ebd730 fixes this